pkg/util/pki/match.go

```diff
@@ -121,15 +132,19 @@ func RequestMatchesSpec(req *cmapi.CertificateRequest, spec cmapi.CertificateSpe
 
 	var violations []string
 
-	if !util.EqualUnsorted(IPAddressesToString(x509req.IPAddresses), spec.IPAddresses) {
+	if !ipSlicesMatch(x509req.IPAddresses, spec.IPAddresses) {
```

Comment:
I definitely agree with this change, for certain `spec.IPAddresses` values, `x509req.IPAddresses` might never match `spec.IPAddresses` because it was parsed into a different format.